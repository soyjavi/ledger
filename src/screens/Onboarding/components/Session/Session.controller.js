import * as LocalAuthentication from 'expo-local-authentication';

import { L10N } from '@common';
import { ServiceNode } from '@services';

export const askLocalAuthentication = async ({
  setPin,
  store: {
    settings: { pin },
  },
}) => {
  const hasHardware = await LocalAuthentication.hasHardwareAsync();
  const isEnrolled = await LocalAuthentication.isEnrolledAsync();

  if (hasHardware && isEnrolled) {
    const { success } = await LocalAuthentication.authenticateAsync({
      promptMessage: L10N.AUTHENTICATE,
      cancelLabel: L10N.CANCEL,
      disableDeviceFallback: true,
    });

    if (success) setPin(pin);
  }
};

export const onHandshake = async ({ onSession, store: { settings, updateSettings } }, pin) => {
  const { fingerprint } = settings;
  const isSignup = settings.pin === undefined;

  if (isSignup)
    await updateSettings({
      pin,
      onboarded: true,
      authorization: await ServiceNode.signup({ fingerprint }),
    });

  onSession();
};
