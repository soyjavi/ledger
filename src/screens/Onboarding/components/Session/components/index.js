import NumKeyboard from './NumKeyboard';

export {
  NumKeyboard, //  eslint-disable-line
};
