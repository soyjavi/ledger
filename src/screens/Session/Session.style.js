import { StyleSheet } from 'react-native';

import { THEME } from 'reactor/common';

const { COLOR, OFFSET } = THEME;

export default StyleSheet.create({
  content: {
    padding: OFFSET,
    paddingTop: '50%',
    backgroundColor: COLOR.PRIMARY,
    width: '100%',
    height: '100%',
  },

  progressBar: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100vw',
  },

  text: {
    color: COLOR.WHITE,
    alignSelf: 'center',
  },
});
