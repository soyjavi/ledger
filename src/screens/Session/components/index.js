import NumKeyboard from './NumKeyboard';
import Pin from './Pin';

export {
  NumKeyboard,
  Pin,
};
