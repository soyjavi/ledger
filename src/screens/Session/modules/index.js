export * from './onHandshake';
export * from './onFingerprint';
export * from './onPin';
