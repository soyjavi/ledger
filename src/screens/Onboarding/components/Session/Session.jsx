import PropTypes from 'prop-types';

import React, { useEffect, useState } from 'react';
import { THEME } from 'reactor/common';
import { Motion, Row, Text, View, Viewport } from 'reactor/components';

import { C } from '@common';
import { useL10N, useStore } from '@context';

import { NumKeyboard } from './components';
import { onHandshake, askLocalAuthentication } from './Session.controller';
import styles from './Session.style';

const { IS_DEV, VERSION } = C;
const { COLOR } = THEME;

const Session = ({ onSession, visible, ...others }) => {
  const l10n = useL10N();
  const store = useStore();

  const [busy, setBusy] = useState(false);
  const [pin, setPin] = useState('');

  const handleHandshake = onHandshake.bind(undefined, { onSession, setBusy, store });

  useEffect(() => {
    if (visible) {
      const { settings, vaults = [] } = store;

      if (settings.pin && vaults.length !== 0) {
        if (IS_DEV /* && IS_WEB */) setPin(settings.pin);
        else if (pin === '') askLocalAuthentication({ l10n, setPin, store });
      }
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [visible]);

  useEffect(() => {
    if (pin.length === 4) {
      const { settings } = store;
      if (settings.pin === undefined || settings.pin === pin) handleHandshake(pin);
      else setPin('');
    }
  }, [pin]);

  const signup = store.authorization === undefined;

  return (
    <Viewport {...others} visible={visible} scroll={false}>
      <View style={styles.container}>
        <Text bold subtitle>
          {signup ? l10n.PIN_CHOOSE : l10n.PIN}
        </Text>
        <Row justify="center">
          {['•', '•', '•', '•'].map((letter, index) => (
            <Motion key={index} timeline={[{ property: 'scale', value: pin.length > index ? 1.5 : 1 }]} type="spring">
              <Text headline color={pin.length <= index ? COLOR.LIGHTEN : undefined}>
                {letter}
              </Text>
            </Motion>
          ))}
        </Row>

        <NumKeyboard onPress={(number) => setPin(`${pin}${number}`)} />
        <Text caption color={COLOR.LIGHTEN} marginBottom="M">{`v${VERSION}`}</Text>
      </View>
    </Viewport>
  );
};

Session.propTypes = {
  onSession: PropTypes.func,
  signup: PropTypes.bool,
  visible: PropTypes.bool,
};

export { Session };
