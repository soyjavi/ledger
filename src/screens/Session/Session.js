import { bool, func } from 'prop-types';
import React, { Fragment, PureComponent } from 'react';
import { Image, View } from 'react-native';

import ASSETS from '../../assets';
import { C } from '../../common';
import { Consumer } from '../../context';
import { THEME } from '../../reactor/common';
import {
  Activity, Motion, Text, Viewport,
} from '../../reactor/components';
import { NumKeyboard } from './components';
import handshake from './modules/handshake';
import styles from './Session.style';

const { VERSION } = C;
const { MOTION: { DURATION } } = THEME;

class Session extends PureComponent {
  static propTypes = {
    getFingerprintAsync: func,
    visible: bool,
  };

  static defaultProps = {
    getFingerprintAsync: undefined,
    visible: true,
  };

  state = {
    busy: false,
    askFingerprint: false,
    pin: '',
  }

  _onFingerprint = async ({ navigation, store }) => {
    const { props: { getFingerprintAsync } } = this;

    this.setState({ askFingerprint: true });
    if (getFingerprintAsync) {
      const { success } = await getFingerprintAsync('Use your fingerprint.');
      if (success) handshake(this, { pin: store.pin, store, navigation });
      else this.setState({ askFingerprint: false });
    }
  }

  _onNumber = ({ number, store, navigation }) => {
    let { state: { pin } } = this;

    pin = `${pin}${number}`;
    this.setState({ pin });

    if (pin.length === 4) {
      setTimeout(() => {
        if (store.pin === undefined || store.pin === pin) handshake(this, { pin, store, navigation });
        else this.setState({ pin: '' });
      }, DURATION / 2);
    }
  }

  render() {
    const {
      _onFingerprint, _onNumber,
      props: { getFingerprintAsync, visible, ...props },
      state: { askFingerprint, busy, pin },
    } = this;
    console.log('<Session>', { askFingerprint, visible, busy, pin });

    return (
      <Viewport {...props} scroll={false} visible={visible}>
        <Consumer>
          { ({ l10n, store, navigation }) => (
            <View style={styles.container}>
              { visible && getFingerprintAsync && !askFingerprint && store.pin
                ? _onFingerprint({ store, navigation }) && <Fragment />
                : undefined }
              <View style={styles.content}>
                <Image source={ASSETS.logo} resizeMode="contain" style={styles.logo} />
                <Text headline level={4} style={styles.text}>voltvault</Text>
                <View style={styles.pin}>
                  { busy
                    ? <Activity size="large" style={styles.activity} />
                    : [1, 2, 3, 4].map(number => (
                      <Motion
                        key={number}
                        style={[styles.bullet, pin.length >= number && styles.bulletActive]}
                        timeline={[{ property: 'scale', value: pin.length >= number ? 1 : 0.8 }]}
                      />
                    ))}
                </View>
                <Text level={2} lighten style={styles.text}>
                  { store.pin && getFingerprintAsync ? l10n.ENTER_PIN_OR_FINGERPRINT : l10n.ENTER_PIN }
                </Text>
              </View>

              { store.pin && !busy && getFingerprintAsync && (
                <Image source={ASSETS.fingerprint} style={styles.fingerprint} />)}
              <NumKeyboard onPress={number => _onNumber({ number, store, navigation })} />
              <Text lighten caption level={3} style={styles.text}>{`v${VERSION}`}</Text>
            </View>
          )}
        </Consumer>
      </Viewport>
    );
  }
}

export default Session;
