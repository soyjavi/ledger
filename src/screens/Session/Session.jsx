import PropTypes from 'prop-types';

import React, { useEffect, useState } from 'react';
import { THEME } from 'reactor/common';
import { Row, Text, View, Viewport } from 'reactor/components';
import { useEnvironment } from 'reactor/hooks';

import { C } from '@common';
import { useL10N, useSnackBar, useStore } from '@context';

import { NumKeyboard } from './components';
import { onHandshake, onFingerprint, onPin } from './modules';
import styles from './Session.style';

const { IS_DEV, VERSION } = C;
const { COLOR } = THEME;

export const Session = ({ onProfile, visible, ...others }) => {
  const { IS_WEB } = useEnvironment();
  const l10n = useL10N();
  const store = useStore();

  const [busy, setBusy] = useState(false);
  const [fingerprint, setFingerprint] = useState(undefined);
  const [pin, setPin] = useState('');

  const handleHandshake = onHandshake.bind(undefined, { onProfile, setBusy, store });
  const handleFingerprint = onFingerprint.bind(undefined, { handleHandshake, setFingerprint, store });
  const handlePin = onPin.bind(undefined, { handleHandshake, setPin, store });

  useEffect(() => {
    const { settings, vaults = [] } = store;

    if (visible && settings.pin && vaults.length !== 0) {
      if (IS_DEV && IS_WEB) handleHandshake(settings.pin);
      else if (fingerprint === undefined) handleFingerprint();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [visible]);

  const signup = store.authorization === undefined;

  return (
    <Viewport {...others} visible={visible} scroll={false}>
      <View style={styles.container}>
        <Text headline>{signup ? l10n.PIN_CHOOSE : l10n.PIN}</Text>
        <Row justify="center" style={styles.input}>
          {!busy ? (
            <>
              {['•', '•', '•', '•'].map((letter, index) => (
                <Text key={index} headline color={pin.length <= index ? COLOR.LIGHTEN : undefined}>
                  {letter}
                </Text>
              ))}
            </>
          ) : (
            <Text>$$ Wait a moment</Text>
          )}
        </Row>

        <NumKeyboard onPress={(number) => handlePin(`${pin}${number}`)} />
        <Text caption color={COLOR.LIGHTEN} marginBottom="M">{`v${VERSION}`}</Text>
      </View>
    </Viewport>
  );
};

Session.propTypes = {
  onProfile: PropTypes.func,
  signup: PropTypes.bool,
  visible: PropTypes.bool,
};
