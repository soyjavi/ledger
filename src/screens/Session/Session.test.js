import React from 'react';
import renderer from 'react-test-renderer';

import { C } from '../../common';
import Session from './Session';

describe('<Session>', () => {
  it('renders', () => {
    const tree = renderer.create(<Session />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
