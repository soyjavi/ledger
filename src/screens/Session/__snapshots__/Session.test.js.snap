// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Session> renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": 1334,
      "left": 0,
      "overflow": "hidden",
      "position": "absolute",
      "top": 0,
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
      "width": 750,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
            "zIndex": 0,
          },
          Array [],
        ]
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": "100%",
            "maxWidth": 400,
            "minWidth": 320,
            "paddingHorizontal": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 64,
                  "marginRight": 8,
                  "marginTop": 8,
                  "width": 64,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#4a4a4a",
                    "fontFamily": undefined,
                    "fontWeight": "400",
                    "textAlignVertical": "center",
                  },
                  Object {
                    "fontFamily": undefined,
                    "fontSize": 24,
                    "fontWeight": "700",
                  },
                  false,
                  Object {
                    "fontSize": 64,
                    "zIndex": 1,
                  },
                ]
              }
            >
              volt.
            </Text>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#cdd2d4",
                  "borderRadius": 10,
                  "height": 20,
                  "margin": 10,
                  "transform": Array [
                    Object {
                      "scale": 0.8,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#cdd2d4",
                  "borderRadius": 10,
                  "height": 20,
                  "margin": 10,
                  "transform": Array [
                    Object {
                      "scale": 0.8,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#cdd2d4",
                  "borderRadius": 10,
                  "height": 20,
                  "margin": 10,
                  "transform": Array [
                    Object {
                      "scale": 0.8,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#cdd2d4",
                  "borderRadius": 10,
                  "height": 20,
                  "margin": 10,
                  "transform": Array [
                    Object {
                      "scale": 0.8,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4a4a4a",
                  "fontFamily": undefined,
                  "fontWeight": "400",
                  "textAlignVertical": "center",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 19.2,
                },
                Object {
                  "color": "#707070",
                },
                Object {},
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "flex-end",
              "bottom": 16,
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                1
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                2
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                3
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                4
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                5
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                6
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                7
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                8
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                9
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="none"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "fontWeight": "400",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {
                      "fontSize": 24,
                    },
                  ]
                }
              >
                0
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="none"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "height": 56,
                  "justifyContent": "center",
                  "minWidth": 214.28571428571428,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 3,
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 28,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#10c0bc",
                    "height": 0,
                    "opacity": 0,
                    "width": 0,
                    "zIndex": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontFamily": undefined,
                "fontWeight": "400",
                "textAlignVertical": "center",
              },
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "#707070",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          v0.18.0
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
